From 8b283e1ed08b236fbf926412356ee9d06533ce9f Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:51:25 -0400
Subject: [tools] perf/scripting/perl: Shorten function signatures

Message-id: <1408953210-26343-198-git-send-email-jolsa@redhat.com>
Patchwork-id: 88885
O-Subject: [PATCH RHEL7.1 BZ1133083 197/322] perf scripting perl: Shorten function signatures
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 8853a1b76288d79aa1a6c5b8fe623d892aa9958b
Author: Arnaldo Carvalho de Melo <acme@redhat.com>
Date: Thu Dec 19 16:39:31 2013 -0300

description
===========
Removing unused parameters.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/scripting-engines/trace-event-perl.c b/tools/perf/util/scripting-engines/trace-event-perl.c
index d5e5969..b672ef0 100644
--- a/tools/perf/util/scripting-engines/trace-event-perl.c
+++ b/tools/perf/util/scripting-engines/trace-event-perl.c
@@ -257,12 +257,9 @@ static inline struct event_format *find_cache_event(struct perf_evsel *evsel)
  return event;
 }
 
-static void perl_process_tracepoint(union perf_event *perf_event __maybe_unused,
-        struct perf_sample *sample,
+static void perl_process_tracepoint(struct perf_sample *sample,
         struct perf_evsel *evsel,
-        struct machine *machine __maybe_unused,
-        struct thread *thread,
-     struct addr_location *al)
+        struct thread *thread)
 {
  struct format_field *field;
  static char handler[256];
@@ -349,10 +346,7 @@ static void perl_process_tracepoint(union perf_event *perf_event __maybe_unused,
 
 static void perl_process_event_generic(union perf_event *event,
            struct perf_sample *sample,
-           struct perf_evsel *evsel,
-           struct machine *machine __maybe_unused,
-           struct thread *thread __maybe_unused,
-        struct addr_location *al __maybe_unused)
+           struct perf_evsel *evsel)
 {
  dSP;
 
@@ -377,12 +371,12 @@ static void perl_process_event_generic(union perf_event *event,
 static void perl_process_event(union perf_event *event,
           struct perf_sample *sample,
           struct perf_evsel *evsel,
-          struct machine *machine,
+          struct machine *machine __maybe_unused,
           struct thread *thread,
-       struct addr_location *al)
+          struct addr_location *al __maybe_unused)
 {
- perl_process_tracepoint(event, sample, evsel, machine, thread, al);
- perl_process_event_generic(event, sample, evsel, machine, thread, al);
+ perl_process_tracepoint(sample, evsel, thread);
+ perl_process_event_generic(event, sample, evsel);
 }
 
 static void run_start_sub(void)
-- 
1.7.1